modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuildControllerTest.java

```diff
@@ -65,7 +65,9 @@
 import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
 import org.apache.ignite.internal.util.IgniteUtils;
 import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.ClusterNodeImpl;
```

Comment:
Seems that given PR doesn't match neither ticket description no ticket caption. Basically you've just added leasholderId to the lease along with corresponding serialization adjustments, refactoring etc. Proposed changes looks good and isolated however it's just a prerequisite to "Find a way to get a MvTableStorage on recovery the index build" and states denoted in ticket description `If during recovery we use PlacementDriver#getPrimaryReplica and get that we are not the primary replica, then we will not build an index, otherwise an honest selection of the primary replica will occur using the replication log.` aren't covered. Please split the ticket and(or) reformulate given one in a way that it's only about adding leasholderId.

Reply:
I corrected the title, removed the description to section “123”, and added what needs to be done in the end.

Reply:
No, it's fine.